src/lib.rs

```diff
@@ -767,6 +870,29 @@ fn parse_version(bytes: &mut Bytes<'_>) -> Result<u8> {
     Ok(Status::Partial)
 }
 
+#[inline]
+#[doc(hidden)]
+#[allow(missing_docs)]
+pub fn parse_method<'a>(bytes: &mut Bytes<'a>) -> Result<&'a str> {
```

Comment:
I think my only last concern is around making things public but undocumented. Even though they are hidden from docs, people inevitably start using them, and then complain if they are changed. I think it's a good idea to solve this with one of two ways:

1. Go through a few extra hoops to only make them public when running the benchmarks. Basically, we could move these functions into a submodule, so that they `pub fn` isn't automatically crate-public. Then, add `#[cfg(httparse_benches)]` to the `mod _benchable`. Running the benches would require passing `RUSTFLAGS="--cfg httparse_benches"`. But accidental usage would be basically nonexistent.
2. Or, embrace "if its not documented, don't use it or we will break you". Perhaps with a code comment at each on of this `doc(hidden)`. People will inevitably use it anyways, but we could bask in the "we told you not to hold it that way". Maybe.

Reply:
I took the latter approach, added to #122 (just benchmarks). If that's good let's land #122, I'll rebase this (#123) on top with a minor cleanup and submit another PR with the aforementioned SIMD improvements for large inputs. Good ?